### ISSUE TITLE
Properly encode an absolute path with percent encoding in `WebClient`

### ISSUE THREAD

**Issue by ikhoon** [line/armeria#3780]
If a relative path is sent, `PathAndQuery.parse()` correctly encode the path.
```java
// '/abc/1 2/def' will be encoded to '/abc/1%202/def'
WebClient.of("http://example.com").get("/abc/1 2/def").aggregate();
```
However, an absolute path containing a whitespace is sent, `URI.create()` will raise
an `URISyntaxException`.
```java
WebClient.of().get("http://example.com/abc/1 2/def").aggregate();
```

**Comment by ikhoon** [comment]
Another problem is that the encoded path(`/abc/1%202/def`) is not updated to `RequestHeaders`.
The following example was reported by @kezhenxu94. https://line-armeria.slack.com/archives/C1NGPBUH2/p1629466435104000?thread_ts=1629462093.100100&cid=C1NGPBUH2
```
mkdir -p p1/p2/p3 
touch p1/p2/p3/"hello world"
python3 -m http.server 8080 
```
```
public static void main(String[] args) throws ExecutionException, InterruptedException {
    final WebClient c = WebClient.of("http://localhost:8080");
    System.out.println(c.execute(HttpRequest.builder().get("/p1/p2/p3/{name}")
                                            .pathParam("name", "hello world")
                                            .build())
                        .aggregate()
                        .get()
                        .status());
}
```
:point_up_2: doesn't work, 400 bad request
:point_down: works, 200 OK
```
public static void main(String[] args) throws ExecutionException, InterruptedException {
    final WebClient c = WebClient.of("http://localhost:8080");
    System.out.println(c.execute(HttpRequest.builder().get("/p1/p2/p3/{name}")
                                            .pathParam("name", "hello%20world")
                                            .build())
                        .aggregate()
                        .get()
                        .status());
}
```

**Comment by ikhoon** [comment]
There is an encoding mismatch on the server-side.
https://github.com/line/armeria/blob/5b384fbe27e7e6f9225d6db91cbb684d09dfbb5e/core/src/test/java/com/linecorp/armeria/client/HttpClientIntegrationTest.java#L319-L325
It should decode percent encoding in both `ServiceRequestContext.path()` and `HttpRequest.headers().path()`

**Comment by ikhoon** [comment]
This was fixed by @trustin's work #4789
I double checked the path component of `"http://example.com/abc/1 2/def"` was converted into `/abc/1%202/def`.
